hpx/util/coroutine/detail/context_posix.hpp

```diff
@@ -209,15 +209,15 @@ namespace hpx { namespace util { namespace coroutines {
          *  a new stack. The stack size can be optionally specified.
          */
         template<typename Functor>
-        explicit ucontext_context_impl(Functor& cb, std::ptrdiff_t stack_size)
+        explicit ucontext_context_impl(Functor & cb, std::ptrdiff_t stack_size)
           : m_stack_size(stack_size == -1 ? (std::ptrdiff_t)default_stack_size : stack_size),
             m_stack(alloc_stack(m_stack_size)),
-            cb_(cb)
+            cb_(&cb)
         {
             HPX_ASSERT(m_stack);
             funp_ = &trampoline<Functor>;
             int error = HPX_COROUTINE_MAKE_CONTEXT(
-                &m_ctx, m_stack, m_stack_size, (void (*)(void*))(funp_), &cb_, NULL);
+                &m_ctx, m_stack, m_stack_size, funp_, cb_, NULL);
```

Comment:
Disregard, `cb_` is already a `void*`, not a reference to the functor as the `cb` constructor argument.